interpreters/wamr/Kconfig

```diff
@@ -70,6 +70,11 @@ config INTERPRETERS_WAMR_LIBC_BUILTIN
 	bool "Enable built-in libc"
 	default n
 
+config INTERPRETERS_WAMR_BUILD_WASM_MODULES_FOR_WAMR_LIBC_BUILTIN
```

Comment:
```suggestion
config INTERPRETERS_WAMR_BUILD_MODULES
```

Reply:
i want to make it clear it's about a non-standard abi, rather than wasi.

Reply:
@yamt what about: INTERPRETERS_WAMR_BUILD_MODULES_NO_STD_ABI ?

Reply:
@yamt ping

Reply:
well, what's wrong with the original name? too long?

Reply:
WASM_BUILD is defined to control whether to build wasm module, why no change the default value of WASM_BUILD to n directly?

Reply:
WASM_BUILD is supposed to be set by each apps, isn't it?
eg. https://github.com/apache/nuttx-apps/blob/ba075747be4bbd8b9a2a2f86a02593d4a5a40413/benchmarks/coremark/Makefile#L135
i'm not sure what you mean by "change the default value of WASM_BUILD to n directly".

Reply:
> WASM_BUILD is supposed to be set by each apps, isn't it? eg.
> 

Yes.

> https://github.com/apache/nuttx-apps/blob/ba075747be4bbd8b9a2a2f86a02593d4a5a40413/benchmarks/coremark/Makefile#L135
> 
> 
> i'm not sure what you mean by "change the default value of WASM_BUILD to n directly".

The build system doesn't generate wasm module unless the user set WASM_BUILD=y in Makefile explicitly. So WASM_BUILD support the same functionality of INTERPRETERS_WAMR_BUILD_WASM_MODULES_FOR_WAMR_LIBC_BUILTIN.

Reply:
> > https://github.com/apache/nuttx-apps/blob/ba075747be4bbd8b9a2a2f86a02593d4a5a40413/benchmarks/coremark/Makefile#L135
> > 
> > i'm not sure what you mean by "change the default value of WASM_BUILD to n directly".
> 
> The build system doesn't generate wasm module unless the user set WASM_BUILD=y in Makefile explicitly. So WASM_BUILD support the same functionality of INTERPRETERS_WAMR_BUILD_WASM_MODULES_FOR_WAMR_LIBC_BUILTIN.

do you mean to remove hardcoded WASM_BUILD from coremark etc?

Reply:
Yes. It's unsuitable to set WASM_BUILD to both in the common config. It's better to add a new coremark specific option to build wasm version.

Reply:
i agree it should be able to control wasm build on per-app basis.

on the other hand, my goal here is a bit different.
i want to disable the whole these wasm build stuff because it's causing problems.

Reply:
but if WASM_BUILD is disabled by default, why do it make trouble to you? It's redundant to request people to enable both INTERPRETERS_WAMR_BUILD_WASM_MODULES_FOR_WAMR_LIBC_BUILTIN and xxx_WASM_MODULE for wasm module.

Reply:
Well, if someone volunteer to do it in a way it fixes #2046, I probably don't object.
But I myself don't volunteer at this point. To me, it isn't obvious how the LINK_WASM part can be conditional on per app config.

Otherwise, please consider merging this PR for now to stop interfering people who is not even interested in this feature.
An extra step to enable it can be considered a good thing as this is a very experimental feature.

Reply:
@no1wudi could you take a look?

Reply:
I will do the further imporvement but as @yamt mentioned, a global configuration to control the behavior is OK, just like `up_enable_irq` and `up_irq_enable`.